capstone/fitness/src/main/java/com/google/sps/util/BananaQueue.java

```diff
@@ -0,0 +1,204 @@
+    return addPeelQueue(peelQueueTag, new PeelQueue());
+  }
+  
+  /**
+   * Adds given PeelQueue to the front element of the BananaQueue with the given tag. 
+   *
+   * @param peelQueueTag Tag to access PeelQueue.
+   * @param peelQueue PeelQueue to add.
+   * @return true if add is successful
+   */
+  public boolean addPeelQueue(String peelQueueTag, PeelQueue peelQueue) {
+    if(head == null || head.peelQueueExists(peelQueueTag)) {
+      return false;
+    }
+
+    head.addPeelQueue(peelQueueTag, peelQueue);
+    return true;
+  }
+
+  /**
+   * Removes PeelQueue from the front element of the BananaQueue with the given tag. 
+   *
+   * @param peelQueueTag Tag to access PeelQueue.
+   * @return removed PeelQueue
+   */
+  public PeelQueue removePeelQueue(String peelQueueTag) {
+    if(head == null || !head.peelQueueExists(peelQueueTag)) {
+      return null;
+    }
+
+    return head.removePeelQueue(peelQueueTag);
+  }
+
+  /**
+   * Adds BananaNode to the end of the queue and increments the size.
+   * 
+   * @param banana BananaNode to add
+   * @return true if enqueue is successful
+   */
+  public boolean enqueueBanana(BananaNode banana) {
+    if(banana == null) {
+      return false;
+    }
+
+    if(head == null) {
+      head = banana;
+    }
+    else { 
+      BananaNode temp = foot != null ? foot : head;
+      foot = banana;
+      temp.setNext(foot);
+      foot.setPrev(temp);
+    }
+
+    size++;
+    return true;
+  }
+  
+  /**
+   * Marks the front BananaNode in the queue as complete, decrements the size, moves the 
+   * front of the queue to the next element (can be null), and copies all the PeelQueues 
+   * in the old front to the new front (if not null).
+   *
+   * @return dequeued BananaNode
+   */
+  public BananaNode dequeueBanana() {
+    if(head == null) {
+      return null;
+    }
+
+    head.setComplete(true);
+    if(head.getNext() != null) {
+      head.getNext().addPeels(head.getPeels());
+    }
+    head.removePeels();
+    BananaNode temp = head;
+    head = head.getNext();
+    size--;
+    return temp;
+  }
+
+  /**
+   * Returns the front BananaNode of the queue (can be null).
+   * 
+   * @return front of BananaQueue.
+   */
+  public BananaNode peekBanana() {
+    return head;
+  }
+
+  /**
+   * Returns the front PeelNode of the queue (can be null) given an associated tag.
+   *
+   * @param peelQueueTag Tag to associated PeelQueue.
+   * @return front of PeelQueue mapped to tag
+   */
+  public PeelNode peekPeel(String peelQueueTag) {
+    if(head == null || !head.peelQueueExists(peelQueueTag)) {
+      return null;
+    }
+    return head.getPeelQueue(peelQueueTag).peek(); 
+  }
+  
+  /**
+   * Returns the number of BananaNodes in the BananaQueue.
+   * 
+   * @return number of BananaNodes in the BananaQueue.
+   */
+  public int getSize() {
+    return size;
+  }
+
+  /**
+   * Returns the number of PeelNodes in the PeelQueue of the associated tag. If not found, 
```

Comment:
Not blocking for this review, but instead of using a number like this, you could consider using Optional (https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html).

Reply:
I added Optional, but I wasn't too sure about the implementation. Is what I put, what you had in mind or should I just be returning the Optional object?

Reply:
I was thinking to return the Optional<Integer> and callers can check isPresent so that you can avoid using -1. Feel free to consider this at some other point though to get this submitted first.

Reply:
done